datadog/tests/resource_datadog_user_test.go

```diff
@@ -114,6 +118,98 @@ func TestAccDatadogUser_NoInvitation(t *testing.T) {
 	})
 }
 
+func TestAccDatadogUser_ServiceAccount(t *testing.T) {
+	t.Parallel()
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	username := strings.ToLower(uniqueEntityName(ctx, t)) + "@example.com"
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogUserV2Destroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogUserConfigRequiredServiceAccount(username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserV2Exists(accProvider, "datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", username),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "verified", "true"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "service_account", "true"),
+					resource.TestCheckNoResourceAttr(
+						"datadog_user.foo", "user_invitation_id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_ServiceAccountForceNew(t *testing.T) {
+	t.Parallel()
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	username := strings.ToLower(uniqueEntityName(ctx, t)) + "@example.com"
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogUserV2Destroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogUserConfigRequiredDisabled(username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserV2Exists(accProvider, "datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", username),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "verified", "false"),
+				),
+			},
+			{
+				Config: testAccCheckDatadogUserConfigRequiredServiceAccountForceNew(username),
```

Comment:
Instead of creating a config for force new, we could reuse the `testAccCheckDatadogUserConfigRequiredServiceAccount`. The config in this function and `testAccCheckDatadogUserConfigRequiredServiceAccountForceNew` is the same except for name.

Reply:
thanks for the help, fixed in a9e7d627